src/Microsoft.AspNetCore.Mvc.ViewFeatures/ViewFeatures/HtmlHelperOptions.cs

```diff
@@ -56,5 +56,12 @@ public string IdAttributeDotReplacement
         /// <see cref="IHtmlHelper.ValidationSummary"/> and other overloads.
         /// </summary>
         public string ValidationSummaryMessageElement { get; set; } = "span";
+
+        /// <summary>
+        /// Gets or sets a value that indicates whether the maxlength attribute should be rendered for compatible HTML input elements,
+        /// when they're bound to models marked with either
+        /// <see cref="System.ComponentModel.DataAnnotations.StringLengthAttribute"/> or <see cref="System.ComponentModel.DataAnnotations.MaxLengthAttribute"/> attributes.
+        /// </summary>
+        public bool MaxLengthAttributeRenderingEnabled { get; set; } = false;
```

Comment:
Leave off `= false;` Also suggest this should be `true` by default when using `CompatibilityVersion.Version_2_2` or `Latest`.